src/ast.c

```diff
@@ -459,8 +461,14 @@ static value_t julia_to_scm_(jl_value_t *v)
         fl_free_gc_handles(1);
         return scmv;
     }
-    if (jl_typeis(v, jl_linenumbernode_type))
-        return julia_to_list2((jl_value_t*)line_sym, jl_fieldref(v,0));
+    if (jl_typeis(v, jl_linenumbernode_type)) {
+        value_t args = julia_to_list2(jl_fieldref(v,1), jl_fieldref(v,0));
```

Comment:
i think you are still missing a gc-root for `args` during the call to `julia_to_scm_`